### PR TITLE
Added required devel packages

### LINKED ISSUE #2922: A few hiccups with tutorial #4
I've experienced a few hiccups going through the tutorial https://github.com/containers/buildah/blob/master/docs/tutorials/04-include-in-your-build-tool.md 

Adding Buildah as a Go dependency "as-is" fails with errors:

```
# github.com/containers/storage/drivers/btrfs
../go/pkg/mod/github.com/containers/storage@v1.24.5/drivers/btrfs/btrfs.go:8:10: fatal error: btrfs/ioctl.h: No such file or directory
 #include <btrfs/ioctl.h>
          ^~~~~~~~~~~~~~~
compilation terminated.
# pkg-config --cflags  -- devmapper
Package devmapper was not found in the pkg-config search path.
Perhaps you should add the directory containing `devmapper.pc'
to the PKG_CONFIG_PATH environment variable
No package 'devmapper' found
pkg-config: exit status 1
# github.com/mtrmac/gpgme
../go/pkg/mod/github.com/mtrmac/gpgme@v0.1.2/data.go:4:11: fatal error: gpgme.h: No such file or directory
 // #include <gpgme.h>
           ^~~~~~~~~
compilation terminated.
```
This is fixed by installing the following packages:
```
zypper in libbtrfs-devel libgpgme-devel device-mapper-devel
```
I believe, mentioning the required packages/libraries would make the tutorial more welcoming for new users.

Some context: I'm using a VitualBox VM with
```
OS: openSUSE Leap 15.2
go version go1.15.6 linux/amd64
```

### PR DESCRIPTION
**Pull request by tvendelin**
Signed-off-by: Toomas Vendelin <toomas.vendelin@gmail.com>

#### What type of PR is this?

/kind documentation

#### Which issue(s) this PR fixes:

Fixes #2922

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Addresses: #2922 
LGTM
I've added the addresses stanza as I'm not sure your "Fixes" will do the auto-close.
Otherwise, TYVM @tvendelin !

File: docs/tutorials/04-include-in-your-build-tool.md [hunk 1]
```diff
@@ -13,14 +13,28 @@ In this tutorial I'll show you how to create a simple CLI tool that creates an i
...
 ```shell
 go mod init
 ```
 
-To initialize the go modules. Now import Buildah as a dependency:
+Next, we should import Buildah as a dependency. However, make sure that you have the following
+developer packages installed:
+
+```shell
+dnf install btrfs-progs-devel gpgme-devel device-mapper-devel
```
Comment (rhatdan) [inline docs/tutorials/04-include-in-your-build-tool.md:27]: Some of these can be avoided via build tags.

Buildah is currently built with
Buildtags = "containers_image_openpgp exclude_graphdriver_devicemapper"

Comment (rhatdan) [reply docs/tutorials/04-include-in-your-build-tool.md:27]: I think you can get rid of btrfs with
CGO_ENABLED=0  go build ...

Comment (tvendelin) [reply docs/tutorials/04-include-in-your-build-tool.md:27]: Well, some might actually use Btrfs. When installing OpenSUSE, Btrfs is the first option in the filesystem drop-down, so someone might even install it by mistake :). Putting a matrix of all possible installation / build scenarios would unnecessarily clutter the tutorial. IMHO, the tutorial should be doable without ugly error messages in the first place. Optimization options could be explored later.

Comment (rhatdan) [reply docs/tutorials/04-include-in-your-build-tool.md:27]: Almost eveyone in the world uses overlay, so while podman can work with btrfs storage backend, it is not well tested.  Podman using overlayfs works fine, The compilation errors are about support for btrfs and devicemapper.
Devicemapper is NOT a supported back end for podman or buildah.

**Pull request merged**